### PR TITLE
requires layout

### PR DESCRIPTION
**Pull request by TerribleDev**
* Set requires layout which will make #47 not needed
* Fix builds on linux by making sure paths are case sensitive
  * Rename `SumoLogic.Logging.NLog/Nlog.config` to  `SumoLogic.Logging.NLog/NLog.config`
  * Fix paths in sln file

### CODE REVIEW

File: SumoLogic.Logging.Log4Net/BufferedSumoLogicAppender.cs [hunk 1]
```diff
@@ -181,6 +181,8 @@ public bool UseConsoleLog
             set; 
         }
 
+        protected override bool RequiresLayout => true;
```
Comment (latkin) [inline SumoLogic.Logging.Log4Net/BufferedSumoLogicAppender.cs]: I'm not against use of newer syntax, but currently this is an isolated major style departure from all the rest of the project. I would request that you maintain the existing code style.

**Comment by latkin** [comment]
LGTM but I don't want to introduce expression-bodied members in isolation like this.

**Comment by TerribleDev** [comment]
thats fair, I've changed it.

**Comment by latkin** [comment]
This LGTM - is there a way to add testing around the desired behavior change?

Not being too expert in these frameworks I don't want to push a change that will break stuff.

**Comment by latkin** [comment]
@TerribleDev I played around with this a bit today, and I'm good with the change. But adding tests is definitely needed.

I have gone ahead and fleshed out the basics of a test case for this. Can you take a look at https://github.com/SumoLogic/sumologic-net-appenders/commit/33fd7329b4dffb20b8ddc5aee16066474f002335 and cherry-pick to your PR branch? Fill in a similar case for the Buffered appender and I'd be happy to merge at that point.

**Comment by TerribleDev** [comment]
Will do

File: SumoLogic.Logging.Log4Net.Tests/BufferedSumoLogicAppenderTest.cs [hunk 2]
```diff
@@ -74,7 +74,7 @@ public void TestSingleMessage()
             Assert.Equal(0, this.messagesHandler.ReceivedRequests.Count);
             Thread.Sleep(TimeSpan.FromMilliseconds(100));
             Assert.Equal(1, this.messagesHandler.ReceivedRequests.Count);
-            Assert.Equal("This is a message\r\n\r\n", this.messagesHandler.LastReceivedRequest.Content.ReadAsStringAsync().Result);
+            Assert.Equal("This is a message" + Environment.NewLine + Environment.NewLine, this.messagesHandler.LastReceivedRequest.Content.ReadAsStringAsync().Result);
```
Comment (TerribleDev) [inline SumoLogic.Logging.Log4Net.Tests/BufferedSumoLogicAppenderTest.cs]: Only windows uses carriage return. Fix these tests so they can run on my ubuntu laptop :)

File: SumoLogic.Logging.Log4Net.Tests/TestErrorHandler.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+using System;
+using log4net.Core;
+
+namespace SumoLogic.Logging.Log4Net.Tests
+{
+    /// <summary>
+    /// Simple test implementation of Log4Net error handler
+    /// </summary>
+    public class TestErrorHandler : IErrorHandler
```
Comment (TerribleDev) [inline SumoLogic.Logging.Log4Net.Tests/TestErrorHandler.cs]: I pulled this class into its own file

File: SumoLogic.Logging.Log4Net.Tests/SumoLogicAppenderTest.cs [hunk 4]
```diff
@@ -94,7 +94,7 @@ public void TestSingleMessage()
             this.log4netLog.Info("This is a message");
 
             Assert.Equal(1, this.messagesHandler.ReceivedRequests.Count);
-            Assert.Equal("-- This is a message\r\n\r\n", this.messagesHandler.LastReceivedRequest.Content.ReadAsStringAsync().Result);
+            Assert.Equal("-- This is a message" + Environment.NewLine + Environment.NewLine, this.messagesHandler.LastReceivedRequest.Content.ReadAsStringAsync().Result);
```
Comment (TerribleDev) [inline SumoLogic.Logging.Log4Net.Tests/SumoLogicAppenderTest.cs]: Again, fix the tests for non-windows OS

**Comment by TerribleDev** [comment]
@latkin pushed

**Comment by latkin** [comment]
Thanks!

**Pull request merged**